bin/fuel-core/src/cli/run/p2p.rs

```diff
@@ -95,6 +95,15 @@ pub struct P2PArgs {
     #[clap(long = "max-connections-per-peer", default_value = "3", env)]
     pub max_connections_per_peer: u32,
 
+    /// Max number of concurrent pending incoming connections
+    /// Useful in mitigating against DDoS attacks
+    #[clap(long = "max-pending-incoming-connections", default_value = "100", env)]
+    pub max_pending_incoming_connections: u32,
+
+    /// Max number of concurrent pending outgoing connections
+    #[clap(long = "max-pending-outgoing-connections", default_value = "100", env)]
+    pub max_pending_outgoing_connections: u32,
```

Comment:
Even though I've added this parameter, I'm not fully sold that it should be present. The rationale is we only dial a peer if: 

a) We are below the desired amount of peers in a gossipsub mesh (which would imply we would be below any potential `max_pending_outgoing_connections` anyway) and we determined that the peer in question would have a reasonable peer score. So essentially our gossipsub config should handle our scenario.

AND 

b) We are relatively certain that peer is reachable i.e. the `multiaddr` for that peer can be successfully dialled - if that is so, we want to prioritize dialling it.

But I may be missing something about the current p2p setup.

Reply:
You are unsure if we should put a limit on outgoing connections?

Reply:
Correct, more specifically on outgoing pending connections (as opposed to established outgoing connections)